internal/datasources/rest/handler.go

```diff
@@ -0,0 +1,207 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/santhosh-tekuri/jsonschema/v6"
+	uritemplate "github.com/std-uritemplate/std-uritemplate/go/v2"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/mindersec/minder/internal/util/schemaupdate"
+	"github.com/mindersec/minder/internal/util/schemavalidate"
+	minderv1 "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+const (
+	// MaxBytesLimit is the maximum number of bytes to read from the response body
+	// We limit to 1MB to prevent abuse
+	MaxBytesLimit int64 = 1 << 20
+)
+
+type restHandler struct {
+	// nis is a nullable input schema
+	rawnis       *structpb.Struct
+	nis          *jsonschema.Schema
+	endpointTmpl string
+	method       string
+	body         string
+	headers      map[string]string
+	parse        string
+	// TODO implement fallback
+	// TODO implement auth
+}
+
+func newHandlerFromDef(def *minderv1.RestDataSource_Def) (*restHandler, error) {
+	if def == nil {
+		return nil, errors.New("rest data source handler definition is nil")
+	}
+
+	// schema may be nil
+	schema, err := schemavalidate.CompileSchemaFromPB(def.GetInputSchema())
+	if err != nil {
+		return nil, err
+	}
+
+	return &restHandler{
+		rawnis:       def.GetInputSchema(),
+		nis:          schema,
+		endpointTmpl: def.GetEndpoint(),
+		method:       defaultString(def.GetMethod(), http.MethodGet),
+		headers:      def.GetHeaders(),
+		body:         parseRequestBodyConfig(def),
+		parse:        def.GetParse(),
+	}, nil
+}
+
+func (h *restHandler) ValidateArgs(args any) error {
+	if h.nis == nil {
+		return nil
+	}
```

Comment:
Does this mean that a data source without a schema accepts _all_ arguments?  It feels like maybe it should accept _no_ arguments.

Reply:
I'm gonna make this mandatory so you always need to add a validation schema.